### DISCUSSION TITLE
are repo and category ID required?

### DISCUSSION THREAD

**Discussion by Andre601** [giscus/giscus#289] [Q&A]
I'm in the process of writing a guide on how to integrate giscus into the Material for MkDocs theme.

Unfortunately does one particular thing cause me some headaches, which is the fact that the `<script>` block uses repository and category IDs.
The issue here is, that I want to explain how to override the default setup on a per-page level. This would, however, require the end-user to also set repo and category ID, which I don't think are easy to obtain under normal circumstances.

Because of this would I like to ask, if the repo and category ID are a required option in the script, or if they can be safely ignored.

**Comment by laymonage** [comment]
Hey, thanks for bringing this up. Here's how it works:
- Repository ID is required to post a new comment, because the GraphQL API requires the repository ID as part of the input (not just `owner/reponame`).
- Category ID is required to automatically create a new discussion if no matching discussion exists.

If you remove the IDs, the above functionalities will not work.

I decided to prefetch the IDs during the setup to avoid making additional calls to the Discussions API. This made more sense when the page wasn't server-rendered, though. Now that the page is server-rendered, I guess it's possible to fetch the IDs from the server-side code if they're not provided by the user. I'll need to update the code to implement the fallback.

**Reply by rimzzlabs** [reply]
Hello, where to get my repository ID and discussion ID? couldn't find any resource that help me to solve this

**Reply by laymonage** [reply]
@rizkimcitra **You need to follow the configuration steps on https://giscus.app. After filling in the repository name, those values will automatically be given to you.**

**Reply by laymonage** [reply]
@stemativablog No, you clearly didn't follow the steps.

## ⚠️ Please, please follow the configuration steps. They're there for a reason.

### 1️⃣ If you input your repository name:

## 👇👇👇

<img width="739" alt="image" src="https://user-images.githubusercontent.com/6379424/174687333-930b4947-7033-458e-a207-827194751c9c.png">

## 👆👆👆

### 2️⃣ And choose the category

## 👇👇👇

<img width="739" alt="image" src="https://user-images.githubusercontent.com/6379424/174687429-eb64b165-5ba9-4cc9-970f-3d3207a6f530.png">

## 👆👆👆

### The ID values will automatically be given to you

## 👇👇👇

<img width="739" alt="image" src="https://user-images.githubusercontent.com/6379424/174687575-66435638-b719-4b62-b68c-642ffb511e56.png">

## 👆👆👆

**Reply by Andre601** [reply]
@laymonage you need to chill a bit... Just link them to the sections jeez...

**Reply by laymonage** [reply]
I don't think linking to the sections will help much as they just scrolled through it. The emojis and headings in that comment were added to separate the text and screenshots, as it can be hard to distinguish the screenshots from GitHub's rendered markdown. I did an extra effort to help them and now I'm told to chill 🤷

**Reply by rimzzlabs** [reply]
I got it now, thank you very much, and sorry for late reply sir, have a great day, cheers to contributors of this repo 😄

**Comment by pawamoy** [comment]
Hi @laymonage, I'm a bit late to the party but:

> I decided to prefetch the IDs during the setup to avoid making additional calls to the Discussions API. This made more sense when the page wasn't server-rendered, though. Now that the page is server-rendered, I guess it's possible to fetch the IDs from the server-side code if they're not provided by the user. I'll need to update the code to implement the fallback.

Is this still relevant today? Can it still be considered? I have described my use-case here if you want to have a look: https://github.com/orgs/giscus/discussions/853 :slightly_smiling_face:
